### ISSUE TITLE
[Android] [Webview] Crash when opening keyboard in webview on specific terminal 

### ISSUE THREAD

**Issue by sawaikatyou** [flutter/flutter#120940]
## Steps to Reproduce

1. Execute `flutter run` on the code sample 
2. Display keyboard in webview. 
3. Leave for about 30 seconds. 

**Expected results:** 
Device won't reboot

**Actual results:** 
Device reboot occurs  
* This issue that occurred only on the F-51B.  
* This issue is not only unique devices.  
  If the model is F-51B, it also occurred on another hardware.
* Bug Occurrence Probability : 7/10


https://user-images.githubusercontent.com/11250845/219525290-d2b968de-d14b-4b07-b5c7-b5b28c3dcdd3.mp4

**Device Information:**  
name: FCNT F-51B  
Android version:12  


<details>
<summary>Code sample</summary>

pubspec.yaml
add dependencies to latest `webview_flutter` & `path_provider`
```yaml
environment:
  sdk: ">=2.17.6 <3.0.0"

# Dependencies specify other packages that your package needs in order to work.
# To automatically upgrade your package dependencies to the latest versions
# consider running `flutter pub upgrade --major-versions`. Alternatively,
# dependencies can be manually updated by changing the version numbers below to
# the latest version available on pub.dev. To see which dependencies have newer
# versions available, run `flutter pub outdated`.
dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^1.0.2
  webview_flutter: ^4.0.2
  path_provider: ^2.0.12
```

android/app/build.gradle
change `minSdkVersion` to 19
```gradle
    defaultConfig {
        minSdkVersion 19 // change
    }
```

main.dart
```dart
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

// ignore_for_file: public_member_api_docs, use_build_context_synchronously, depend_on_referenced_packages

import 'dart:async';
import 'dart:convert';
import 'dart:io';
import 'dart:typed_data';

import 'package:flutter/material.dart';
import 'package:path_provider/path_provider.dart';
import 'package:webview_flutter/webview_flutter.dart';
// #docregion platform_imports
// Import for Android features.
import 'package:webview_flutter_android/webview_flutter_android.dart';
// Import for iOS features.
import 'package:webview_flutter_wkwebview/webview_flutter_wkwebview.dart';
// #enddocregion platform_imports

void main() => runApp(const MaterialApp(home: WebViewExample()));

const String kNavigationExamplePage = '''
<!DOCTYPE html><html>
<head><title>Navigation Delegate Example</title></head>
<body>
<p>
The navigation delegate is set to block navigation to the youtube website.
</p>
<ul>
<ul><a href="https://www.youtube.com/">https://www.youtube.com/</a></ul>
<ul><a href="https://www.google.com/">https://www.google.com/</a></ul>
</ul>
</body>
</html>
''';

const String kLocalExamplePage = '''
<!DOCTYPE html>
<html lang="en">
<head>
<title>Load file or HTML string example</title>
</head>
<body>

<h1>Local demo page</h1>
<p>
  This is an example page used to demonstrate how to load a local file or HTML
  string using the <a href="https://pub.dev/packages/webview_flutter">Flutter
  webview</a> plugin.
</p>

</body>
</html>
''';

const String kTransparentBackgroundPage = '''
  <!DOCTYPE html>
  <html>
  <head>
    <title>Transparent background test</title>
  </head>
  <style type="text/css">
    body { background: transparent; margin: 0; padding: 0; }
    #container { position: relative; margin: 0; padding: 0; width: 100vw; height: 100vh; }
    #shape { background: red; width: 200px; height: 200px; margin: 0; padding: 0; position: absolute; top: calc(50% - 100px); left: calc(50% - 100px); }
    p { text-align: center; }
  </style>
  <body>
    <div id="container">
      <p>Transparent background test</p>
      <div id="shape"></div>
    </div>
  </body>
  </html>
''';

class WebViewExample extends StatefulWidget {
  const WebViewExample({super.key});

  @override
  State<WebViewExample> createState() => _WebViewExampleState();
}

class _WebViewExampleState extends State<WebViewExample> {
  late final WebViewController _controller;

  @override
  void initState() {
    super.initState();

    // #docregion platform_features
    late final PlatformWebViewControllerCreationParams params;
    if (WebViewPlatform.instance is WebKitWebViewPlatform) {
      params = WebKitWebViewControllerCreationParams(
        allowsInlineMediaPlayback: true,
        mediaTypesRequiringUserAction: const <PlaybackMediaTypes>{},
      );
    } else {
      params = const PlatformWebViewControllerCreationParams();
    }

    final WebViewController controller =
    WebViewController.fromPlatformCreationParams(params);
    // #enddocregion platform_features

    controller
      ..setJavaScriptMode(JavaScriptMode.unrestricted)
      ..setBackgroundColor(const Color(0x00000000))
      ..setNavigationDelegate(
        NavigationDelegate(
          onProgress: (int progress) {
            debugPrint('WebView is loading (progress : $progress%)');
          },
          onPageStarted: (String url) {
            debugPrint('Page started loading: $url');
          },
          onPageFinished: (String url) {
            debugPrint('Page finished loading: $url');
          },
          onWebResourceError: (WebResourceError error) {
            debugPrint('''
Page resource error:
  code: ${error.errorCode}
  description: ${error.description}
  errorType: ${error.errorType}
  isForMainFrame: ${error.isForMainFrame}
          ''');
          },
          onNavigationRequest: (NavigationRequest request) {
            if (request.url.startsWith('https://www.youtube.com/')) {
              debugPrint('blocking navigation to ${request.url}');
              return NavigationDecision.prevent;
            }
            debugPrint('allowing navigation to ${request.url}');
            return NavigationDecision.navigate;
          },
        ),
      )
      ..addJavaScriptChannel(
        'Toaster',
        onMessageReceived: (JavaScriptMessage message) {
          ScaffoldMessenger.of(context).showSnackBar(
            SnackBar(content: Text(message.message)),
          );
        },
      )
      ..loadRequest(Uri.parse('https://flutter.dev'));

    // #docregion platform_features
    if (controller.platform is AndroidWebViewController) {
      AndroidWebViewController.enableDebugging(true);
      (controller.platform as AndroidWebViewController)
          .setMediaPlaybackRequiresUserGesture(false);
    }
    // #enddocregion platform_features

    _controller = controller;
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: Colors.green,
      appBar: AppBar(
        title: const Text('Flutter WebView example'),
        // This drop down menu demonstrates that Flutter widgets can be shown over the web view.
        actions: <Widget>[
          NavigationControls(webViewController: _controller),
          SampleMenu(webViewController: _controller),
        ],
      ),
      body: WebViewWidget(controller: _controller),
      floatingActionButton: favoriteButton(),
    );
  }

  Widget favoriteButton() {
    return FloatingActionButton(
      onPressed: () async {
        final String? url = await _controller.currentUrl();
        ScaffoldMessenger.of(context).showSnackBar(
          SnackBar(content: Text('Favorited $url')),
        );
      },
      child: const Icon(Icons.favorite),
    );
  }
}

enum MenuOptions {
  showUserAgent,
  listCookies,
  clearCookies,
  addToCache,
  listCache,
  clearCache,
  navigationDelegate,
  doPostRequest,
  loadLocalFile,
  loadFlutterAsset,
  loadHtmlString,
  transparentBackground,
  setCookie,
}

class SampleMenu extends StatelessWidget {
  SampleMenu({
  super.key,
  required this.webViewController,
  });

  final WebViewController webViewController;
  late final WebViewCookieManager cookieManager = WebViewCookieManager();

  @override
  Widget build(BuildContext context) {
    return PopupMenuButton<MenuOptions>(
      key: const ValueKey<String>('ShowPopupMenu'),
      onSelected: (MenuOptions value) {
        switch (value) {
          case MenuOptions.showUserAgent:
            _onShowUserAgent();
            break;
          case MenuOptions.listCookies:
            _onListCookies(context);
            break;
          case MenuOptions.clearCookies:
            _onClearCookies(context);
            break;
          case MenuOptions.addToCache:
            _onAddToCache(context);
            break;
          case MenuOptions.listCache:
            _onListCache();
            break;
          case MenuOptions.clearCache:
            _onClearCache(context);
            break;
          case MenuOptions.navigationDelegate:
            _onNavigationDelegateExample();
            break;
          case MenuOptions.doPostRequest:
            _onDoPostRequest();
            break;
          case MenuOptions.loadLocalFile:
            _onLoadLocalFileExample();
            break;
          case MenuOptions.loadFlutterAsset:
            _onLoadFlutterAssetExample();
            break;
          case MenuOptions.loadHtmlString:
            _onLoadHtmlStringExample();
            break;
          case MenuOptions.transparentBackground:
            _onTransparentBackground();
            break;
          case MenuOptions.setCookie:
            _onSetCookie();
            break;
        }
      },
      itemBuilder: (BuildContext context) => <PopupMenuItem<MenuOptions>>[
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.showUserAgent,
          child: Text('Show user agent'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.listCookies,
          child: Text('List cookies'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.clearCookies,
          child: Text('Clear cookies'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.addToCache,
          child: Text('Add to cache'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.listCache,
          child: Text('List cache'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.clearCache,
          child: Text('Clear cache'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.navigationDelegate,
          child: Text('Navigation Delegate example'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.doPostRequest,
          child: Text('Post Request'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.loadHtmlString,
          child: Text('Load HTML string'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.loadLocalFile,
          child: Text('Load local file'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.loadFlutterAsset,
          child: Text('Load Flutter Asset'),
        ),
        const PopupMenuItem<MenuOptions>(
          key: ValueKey<String>('ShowTransparentBackgroundExample'),
          value: MenuOptions.transparentBackground,
          child: Text('Transparent background example'),
        ),
        const PopupMenuItem<MenuOptions>(
          value: MenuOptions.setCookie,
          child: Text('Set cookie'),
        ),
      ],
    );
  }

  Future<void> _onShowUserAgent() {
    // Send a message with the user agent string to the Toaster JavaScript channel we registered
    // with the WebView.
    return webViewController.runJavaScript(
      'Toaster.postMessage("User Agent: " + navigator.userAgent);',
    );
  }

  Future<void> _onListCookies(BuildContext context) async {
    final String cookies = await webViewController
        .runJavaScriptReturningResult('document.cookie') as String;
    ScaffoldMessenger.of(context).showSnackBar(SnackBar(
      content: Column(
        mainAxisAlignment: MainAxisAlignment.end,
        mainAxisSize: MainAxisSize.min,
        children: <Widget>[
          const Text('Cookies:'),
          _getCookieList(cookies),
        ],
      ),
    ));
  }

  Future<void> _onAddToCache(BuildContext context) async {
    await webViewController.runJavaScript(
      'caches.open("test_caches_entry"); localStorage["test_localStorage"] = "dummy_entry";',
    );
    ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
      content: Text('Added a test entry to cache.'),
    ));
  }

  Future<void> _onListCache() {
    return webViewController.runJavaScript('caches.keys()'
    // ignore: missing_whitespace_between_adjacent_strings
        '.then((cacheKeys) => JSON.stringify({"cacheKeys" : cacheKeys, "localStorage" : localStorage}))'
        '.then((caches) => Toaster.postMessage(caches))');
  }

  Future<void> _onClearCache(BuildContext context) async {
    await webViewController.clearCache();
    await webViewController.clearLocalStorage();
    ScaffoldMessenger.of(context).showSnackBar(const SnackBar(
      content: Text('Cache cleared.'),
    ));
  }

  Future<void> _onClearCookies(BuildContext context) async {
    final bool hadCookies = await cookieManager.clearCookies();
    String message = 'There were cookies. Now, they are gone!';
    if (!hadCookies) {
      message = 'There are no cookies.';
    }
    ScaffoldMessenger.of(context).showSnackBar(SnackBar(
      content: Text(message),
    ));
  }

  Future<void> _onNavigationDelegateExample() {
    final String contentBase64 = base64Encode(
      const Utf8Encoder().convert(kNavigationExamplePage),
    );
    return webViewController.loadRequest(
      Uri.parse('data:text/html;base64,$contentBase64'),
    );
  }

  Future<void> _onSetCookie() async {
    await cookieManager.setCookie(
      const WebViewCookie(
        name: 'foo',
        value: 'bar',
        domain: 'httpbin.org',
        path: '/anything',
      ),
    );
    await webViewController.loadRequest(Uri.parse(
      'https://httpbin.org/anything',
    ));
  }

  Future<void> _onDoPostRequest() {
    return webViewController.loadRequest(
      Uri.parse('https://httpbin.org/post'),
      method: LoadRequestMethod.post,
      headers: <String, String>{'foo': 'bar', 'Content-Type': 'text/plain'},
      body: Uint8List.fromList('Test Body'.codeUnits),
    );
  }

  Future<void> _onLoadLocalFileExample() async {
    final String pathToIndex = await _prepareLocalFile();
    await webViewController.loadFile(pathToIndex);
  }

  Future<void> _onLoadFlutterAssetExample() {
    return webViewController.loadFlutterAsset('assets/www/index.html');
  }

  Future<void> _onLoadHtmlStringExample() {
    return webViewController.loadHtmlString(kLocalExamplePage);
  }

  Future<void> _onTransparentBackground() {
    return webViewController.loadHtmlString(kTransparentBackgroundPage);
  }

  Widget _getCookieList(String cookies) {
    if (cookies == '""') {
      return Container();
    }
    final List<String> cookieList = cookies.split(';');
    final Iterable<Text> cookieWidgets =
    cookieList.map((String cookie) => Text(cookie));
    return Column(
      mainAxisAlignment: MainAxisAlignment.end,
      mainAxisSize: MainAxisSize.min,
      children: cookieWidgets.toList(),
    );
  }

  static Future<String> _prepareLocalFile() async {
    final String tmpDir = (await getTemporaryDirectory()).path;
    final File indexFile = File(
        <String>{tmpDir, 'www', 'index.html'}.join(Platform.pathSeparator));

    await indexFile.create(recursive: true);
    await indexFile.writeAsString(kLocalExamplePage);

    return indexFile.path;
  }
}

class NavigationControls extends StatelessWidget {
  const NavigationControls({super.key, required this.webViewController});

  final WebViewController webViewController;

  @override
  Widget build(BuildContext context) {
    return Row(
      children: <Widget>[
        IconButton(
          icon: const Icon(Icons.arrow_back_ios),
          onPressed: () async {
            if (await webViewController.canGoBack()) {
              await webViewController.goBack();
            } else {
              ScaffoldMessenger.of(context).showSnackBar(
                const SnackBar(content: Text('No back history item')),
              );
              return;
            }
          },
        ),
        IconButton(
          icon: const Icon(Icons.arrow_forward_ios),
          onPressed: () async {
            if (await webViewController.canGoForward()) {
              await webViewController.goForward();
            } else {
              ScaffoldMessenger.of(context).showSnackBar(
                const SnackBar(content: Text('No forward history item')),
              );
              return;
            }
          },
        ),
        IconButton(
          icon: const Icon(Icons.replay),
          onPressed: () => webViewController.reload(),
        ),
      ],
    );
  }
}
```

</details>

<details>
  <summary>Device Logs</summary>

<!--
      Run your application with `flutter run --verbose` and attach all the
      log output below between the lines with the backticks. If there is an
      exception, please see if the error message includes enough information
      to explain how to solve the issue. If there is a crash, please include
      the entirety of the crash log including the backtrace.
-->

```

02-16 14:31:36.529 31891 31928 I Watchdog: monitor foreground thread
--------- beginning of crash
02-16 14:31:36.541 31739 31741 F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 31741 (Binder:31739_1), pid 31739 (surfaceflinger)
02-16 14:31:36.640 31739   876 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.640 31739 31742 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.642 31739 31963 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.642 31739 31962 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.642 13097 13097 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstoneProto
02-16 14:31:36.655   686   686 I tombstoned: received crash request for pid 31741
02-16 14:31:36.658 13097 13097 I crash_dump64: performing dump of process 31739 (target tid = 31741)
02-16 14:31:36.680 13097 13097 E DEBUG   : failed to read /proc/uptime: Permission denied
02-16 14:31:36.885 11567 11604 I Finsky  : [34] mwf.run(22): Stats for Executor: BlockingExecutor ncq@f5240e2[Running, pool size = 2, active threads = 0, queued tasks = 0, completed tasks = 13]
02-16 14:31:36.886 11567 11604 I Finsky  : [34] mwf.run(22): Stats for Executor: LightweightExecutor ncq@696ac73[Running, pool size = 4, active threads = 0, queued tasks = 0, completed tasks = 86]
02-16 14:31:36.913 31732 31855 D audio_hw_primary: thread_switch_poll: screen_off : 0, adev->no_output_stream : 1,adev->mode : 0 headset_insert : 0 headset_det : 0
02-16 14:31:36.913 31732 31855 D audio_hw_primary: thread_switch_poll: poll
02-16 14:31:36.914 31732 31855 D audio_hw_primary: thread_switch_poll: set switch 1
02-16 14:31:36.900     0     0 W         : EAR_DET_EN = 1 , EAR_DET_IN = 0
02-16 14:31:36.901     0     0 I switch_state: 1
02-16 14:31:36.942     0     0 W         : EAR_DET_EN = 0 , EAR_DET_IN = 0
02-16 14:31:36.943     0     0 W         : EAR_DET_EN = 0 , EAR_DET_IN = 0
02-16 14:31:37.011 12704 12727 E FrameEvents: updateAcquireFence: Did not find frame.
02-16 14:31:37.070 11567 11604 I Finsky  : [34] mwf.run(22): Stats for Executor: bgExecutor ncq@aa4a730[Running, pool size = 4, active threads = 0, queued tasks = 0, completed tasks = 202]
02-16 14:31:37.096 13097 13097 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
02-16 14:31:37.096 13097 13097 F DEBUG   : Build fingerprint: 'FCNT/F51B/F51B:12/V47RD64B/F51B.20221117:user/release-keys'
02-16 14:31:37.096 13097 13097 F DEBUG   : Revision: '0'
02-16 14:31:37.096 13097 13097 F DEBUG   : ABI: 'arm64'
02-16 14:31:37.097 13097 13097 F DEBUG   : Timestamp: 2023-02-16 14:31:36.679075840+0900
02-16 14:31:37.097 13097 13097 F DEBUG   : Process uptime: 0s
02-16 14:31:37.097 13097 13097 F DEBUG   : Cmdline: /system/bin/surfaceflinger
02-16 14:31:37.097 13097 13097 F DEBUG   : pid: 31739, tid: 31741, name: Binder:31739_1  >>> /system/bin/surfaceflinger <<<
02-16 14:31:37.097 13097 13097 F DEBUG   : uid: 1000
02-16 14:31:37.097 13097 13097 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
02-16 14:31:37.097 13097 13097 F DEBUG   :     x0  0000000000000000  x1  0000000000007bfd  x2  0000000000000006  x3  000000772f7e7d60
02-16 14:31:37.097 13097 13097 F DEBUG   :     x4  0000000000000031  x5  0000000000000031  x6  0000000000000031  x7  0000000000000037
02-16 14:31:37.097 13097 13097 F DEBUG   :     x8  00000000000000f0  x9  15427fe59b52cade  x10 0000000000000000  x11 ffffff80ffffffdf
02-16 14:31:37.097 13097 13097 F DEBUG   :     x12 0000000000000001  x13 0000000000000004  x14 0000000000000010  x15 0000000000000000
02-16 14:31:37.097 13097 13097 F DEBUG   :     x16 0000007a40be7050  x17 0000007a40bc40a0  x18 000000772f598000  x19 0000000000007bfb
02-16 14:31:37.097 13097 13097 F DEBUG   :     x20 0000000000007bfd  x21 00000000ffffffff  x22 000001bf95091c64  x23 800001bf7927b9fa
02-16 14:31:37.097 13097 13097 F DEBUG   :     x24 0000007a42672d50  x25 0000007a42670818  x26 0000007a42670818  x27 0000000000000001
02-16 14:31:37.097 13097 13097 F DEBUG   :     x28 000000772f7e8160  x29 000000772f7e7de0
02-16 14:31:37.097 13097 13097 F DEBUG   :     lr  0000007a40b7566c  sp  000000772f7e7d40  pc  0000007a40b75698  pst 0000000000001000
02-16 14:31:37.097 13097 13097 F DEBUG   : backtrace:
02-16 14:31:37.097 13097 13097 F DEBUG   :       #00 pc 0000000000051698  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168) (BuildId: b94752b6809cd7518c3c28ff9d195306)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #01 pc 0000000000038ef8  /system_ext/lib64/libcomposerextn.qti.so (smomo::BufferTracker::IsBufferLate(long, long, long, long)+884) (BuildId: 0a10c9cc500ee91d45b0d2d080c482eb)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #02 pc 000000000003a860  /system_ext/lib64/libcomposerextn.qti.so (smomo::FreeRunSyncState::FrameIsLate(smomo::PresentArgs&)+232) (BuildId: 0a10c9cc500ee91d45b0d2d080c482eb)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #03 pc 0000000000044ae0  /system_ext/lib64/libcomposerextn.qti.so (smomo::SmomoImpl::FrameIsLate(int, long)+808) (BuildId: 0a10c9cc500ee91d45b0d2d080c482eb)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #04 pc 00000000001be204  /system/bin/surfaceflinger (_ZNSt3__110__function6__funcIZN7android14SurfaceFlinger19setTransactionStateERKNS2_17FrameTimelineInfoERKNS2_6VectorINS2_13ComposerStateEEERKNS7_INS2_12DisplayStateEEEjRKNS2_2spINS2_7IBinderEEERKNS2_19InputWindowCommandsElbRKNS2_14client_cache_tEbRKNS_6vectorINS2_17ListenerCallbacksENS_9allocatorISS_EEEEmE4$_64NST_ISY_EEFvRKNS2_13layer_state_tEEEclES12_$316ab8feed68e3a1dff713cb63dee1c6+240) (BuildId: d26adbd9527290a0292af3d634a1b11c)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #05 pc 000000000019f390  /system/bin/surfaceflinger (android::SurfaceFlinger::setTransactionState(android::FrameTimelineInfo const&, android::Vector<android::ComposerState> const&, android::Vector<android::DisplayState> const&, unsigned int, android::sp<android::IBinder> const&, android::InputWindowCommands const&, long, bool, android::client_cache_t const&, bool, std::__1::vector<android::ListenerCallbacks, std::__1::allocator<android::ListenerCallbacks> > const&, unsigned long)+3176) (BuildId: d26adbd9527290a0292af3d634a1b11c)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #06 pc 00000000000c37c0  /system/lib64/libgui.so (android::BnSurfaceComposer::onTransact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+13468) (BuildId: 19476f020d4c592c5bf1891062fc5de3)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #07 pc 00000000001a85a0  /system/bin/surfaceflinger (android::SurfaceFlinger::onTransact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+820) (BuildId: d26adbd9527290a0292af3d634a1b11c)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #08 pc 000000000004a84c  /system/lib64/libbinder.so (android::BBinder::transact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+252) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #09 pc 0000000000054764  /system/lib64/libbinder.so (android::IPCThreadState::executeCommand(int)+1028) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #10 pc 0000000000054294  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+164) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #11 pc 0000000000054b50  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+72) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #12 pc 00000000000810d4  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+28) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #13 pc 0000000000013454  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+264) (BuildId: 5a130e6819cc6e1c6098896371429dc2)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #14 pc 0000000000012cc4  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+408) (BuildId: 5a130e6819cc6e1c6098896371429dc2)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #15 pc 00000000000b508c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+264) (BuildId: b94752b6809cd7518c3c28ff9d195306)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #16 pc 0000000000052f9c  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+68) (BuildId: b94752b6809cd7518c3c28ff9d195306)
02-16 14:31:37.112 31891 32100 W NativeCrashListener: Couldn't find ProcessRecord for pid 31739
```

<!--
     Run `flutter analyze` and attach any output of that command below.
     If there are any analysis errors, try resolving them before filing this issue.
-->

```
$ flutter analyze
Analyzing webview_flutter_test...                                               
No issues found! (ran in 1.2s)
```
</details>

<details>
  <summary>Flutter doctor -v</summary>

<!-- Finally, paste the output of running `flutter doctor -v` here. -->

```
flutter doctor -v                
[✓] Flutter (Channel stable, 3.7.3, on macOS 12.6.1 21G217 darwin-arm64, locale ja-JP)
    • Flutter version 3.7.3 on channel stable at /Users/user/fvm/versions/3.7.3
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 9944297138 (7 days ago), 2023-02-08 15:46:04 -0800
    • Engine revision 248290d6d5
    • Dart version 2.19.2
    • DevTools version 2.20.1
                                                                                                                                                                                                         Checking Android licenses is taking an unexpectedly long time...[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14B47b
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio
    • Android Studio at /Applications/Android Studio Preview.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code (version 1.69.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.46.0

[✓] Proxy Configuration
    • HTTP_PROXY is set
    • NO_PROXY is localhost,::1,127.0.0.1
    • NO_PROXY contains localhost
    • NO_PROXY contains 127.0.0.1
    • NO_PROXY contains ::1

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.6.1 21G217 darwin-arm64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 109.0.5414.119

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

</details>

**Comment by sawaikatyou** [comment]
The crash occurred only once when TextFied was selected with the following code (1/10)

<details>
<summary>Code sample</summary>

main.dart
```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(primarySwatch: Colors.blue),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;
  late TextEditingController _textEditingController;

  @override
  void initState() {
    super.initState();
    _textEditingController = TextEditingController(text: 'input message');
  }

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(title: Text(widget.title)),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            TextField(
                controller: _textEditingController,
                keyboardType: TextInputType.text,
                decoration: const InputDecoration(
                  border: UnderlineInputBorder(),
                  hintText: '',
                  labelText: 'label',
                )),
            const Text('You have pushed the button this many times:'),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}
```
</details>

**Comment by huycozy** [comment]
Hi @sawaikatyou 
Can you check if the issue also happens when running this [simple example](https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter/example/lib/simple_example.dart)? Please also provide the complete output of `flutter run -v` so that we may verify the issue.

>The crash occurred only once when TextFied was selected with the following code (1/10)

Can you be more specific about this? As I see, this is a separated sample code with the above (in OP).

**Comment by sawaikatyou** [comment]
hi @huycozy 
Please wait while we confirm

**Comment by sawaikatyou** [comment]
> Can you check if the issue also happens when running this [simple example](https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter/example/lib/simple_example.dart)? Please also provide the complete output of flutter run -v so that we may verify the issue.

Here is the log [log.txt](https://github.com/flutter/flutter/files/10790826/log.txt)

> Can you be more specific about this? As I see, this is a separated sample code with the above (in OP).

Since this issue crashed while opening the soft keyboard, will it also crash with the TextField Widget? I prepared it because I wanted to find out.
It may be different from the issue I created, but I wrote it as one of the investigations.

**Comment by sawaikatyou** [comment]
> If it also happens with single TextField widget, it seems to be different issue. I would recommend opening new issue for that problem.

I'm looking for a condition where the crash occurs on a single TextField Widget.
It doesn't happen just by placing a TextField in the application created with `flutter create`.
I will create a new issue when I find it.

> For the crash issue with webview plugin, can you try open another website that contains text field so that you can check if there is a crash too? (I don't have a such device so can not check this. It may be a device-specific device issue if it also happens)

I checked if the crash happens on another website.

occured (3/3)
https://www.google.com/
https://www.yahoo.com/
https://github.com/

not occured (0/20)
https://www.bing.com/

> Also, can you try running this Android Native WebView sample code to check if there is any crash?

I checked and it didn't happen. (0/30)

**Comment by huycozy** [comment]
```

02-16 14:31:36.529 31891 31928 I Watchdog: monitor foreground thread
--------- beginning of crash
02-16 14:31:36.541 31739 31741 F libc    : Fatal signal 6 (SIGABRT), code -1 (SI_QUEUE) in tid 31741 (Binder:31739_1), pid 31739 (surfaceflinger)
02-16 14:31:36.640 31739   876 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.640 31739 31742 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.642 31739 31963 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.642 31739 31962 E IPCThreadState: *** IOCTL ERROR errno=4
02-16 14:31:36.642 13097 13097 I crash_dump64: obtaining output fd from tombstoned, type: kDebuggerdTombstoneProto
02-16 14:31:36.655   686   686 I tombstoned: received crash request for pid 31741
02-16 14:31:36.658 13097 13097 I crash_dump64: performing dump of process 31739 (target tid = 31741)
02-16 14:31:36.680 13097 13097 E DEBUG   : failed to read /proc/uptime: Permission denied
02-16 14:31:36.885 11567 11604 I Finsky  : [34] mwf.run(22): Stats for Executor: BlockingExecutor ncq@f5240e2[Running, pool size = 2, active threads = 0, queued tasks = 0, completed tasks = 13]
02-16 14:31:36.886 11567 11604 I Finsky  : [34] mwf.run(22): Stats for Executor: LightweightExecutor ncq@696ac73[Running, pool size = 4, active threads = 0, queued tasks = 0, completed tasks = 86]
02-16 14:31:36.913 31732 31855 D audio_hw_primary: thread_switch_poll: screen_off : 0, adev->no_output_stream : 1,adev->mode : 0 headset_insert : 0 headset_det : 0
02-16 14:31:36.913 31732 31855 D audio_hw_primary: thread_switch_poll: poll
02-16 14:31:36.914 31732 31855 D audio_hw_primary: thread_switch_poll: set switch 1
02-16 14:31:36.900     0     0 W         : EAR_DET_EN = 1 , EAR_DET_IN = 0
02-16 14:31:36.901     0     0 I switch_state: 1
02-16 14:31:36.942     0     0 W         : EAR_DET_EN = 0 , EAR_DET_IN = 0
02-16 14:31:36.943     0     0 W         : EAR_DET_EN = 0 , EAR_DET_IN = 0
02-16 14:31:37.011 12704 12727 E FrameEvents: updateAcquireFence: Did not find frame.
02-16 14:31:37.070 11567 11604 I Finsky  : [34] mwf.run(22): Stats for Executor: bgExecutor ncq@aa4a730[Running, pool size = 4, active threads = 0, queued tasks = 0, completed tasks = 202]
02-16 14:31:37.096 13097 13097 F DEBUG   : *** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
02-16 14:31:37.096 13097 13097 F DEBUG   : Build fingerprint: 'FCNT/F51B/F51B:12/V47RD64B/F51B.20221117:user/release-keys'
02-16 14:31:37.096 13097 13097 F DEBUG   : Revision: '0'
02-16 14:31:37.096 13097 13097 F DEBUG   : ABI: 'arm64'
02-16 14:31:37.097 13097 13097 F DEBUG   : Timestamp: 2023-02-16 14:31:36.679075840+0900
02-16 14:31:37.097 13097 13097 F DEBUG   : Process uptime: 0s
02-16 14:31:37.097 13097 13097 F DEBUG   : Cmdline: /system/bin/surfaceflinger
02-16 14:31:37.097 13097 13097 F DEBUG   : pid: 31739, tid: 31741, name: Binder:31739_1  >>> /system/bin/surfaceflinger <<<
02-16 14:31:37.097 13097 13097 F DEBUG   : uid: 1000
02-16 14:31:37.097 13097 13097 F DEBUG   : signal 6 (SIGABRT), code -1 (SI_QUEUE), fault addr --------
02-16 14:31:37.097 13097 13097 F DEBUG   :     x0  0000000000000000  x1  0000000000007bfd  x2  0000000000000006  x3  000000772f7e7d60
02-16 14:31:37.097 13097 13097 F DEBUG   :     x4  0000000000000031  x5  0000000000000031  x6  0000000000000031  x7  0000000000000037
02-16 14:31:37.097 13097 13097 F DEBUG   :     x8  00000000000000f0  x9  15427fe59b52cade  x10 0000000000000000  x11 ffffff80ffffffdf
02-16 14:31:37.097 13097 13097 F DEBUG   :     x12 0000000000000001  x13 0000000000000004  x14 0000000000000010  x15 0000000000000000
02-16 14:31:37.097 13097 13097 F DEBUG   :     x16 0000007a40be7050  x17 0000007a40bc40a0  x18 000000772f598000  x19 0000000000007bfb
02-16 14:31:37.097 13097 13097 F DEBUG   :     x20 0000000000007bfd  x21 00000000ffffffff  x22 000001bf95091c64  x23 800001bf7927b9fa
02-16 14:31:37.097 13097 13097 F DEBUG   :     x24 0000007a42672d50  x25 0000007a42670818  x26 0000007a42670818  x27 0000000000000001
02-16 14:31:37.097 13097 13097 F DEBUG   :     x28 000000772f7e8160  x29 000000772f7e7de0
02-16 14:31:37.097 13097 13097 F DEBUG   :     lr  0000007a40b7566c  sp  000000772f7e7d40  pc  0000007a40b75698  pst 0000000000001000
02-16 14:31:37.097 13097 13097 F DEBUG   : backtrace:
02-16 14:31:37.097 13097 13097 F DEBUG   :       #00 pc 0000000000051698  /apex/com.android.runtime/lib64/bionic/libc.so (abort+168) (BuildId: b94752b6809cd7518c3c28ff9d195306)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #01 pc 0000000000038ef8  /system_ext/lib64/libcomposerextn.qti.so (smomo::BufferTracker::IsBufferLate(long, long, long, long)+884) (BuildId: 0a10c9cc500ee91d45b0d2d080c482eb)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #02 pc 000000000003a860  /system_ext/lib64/libcomposerextn.qti.so (smomo::FreeRunSyncState::FrameIsLate(smomo::PresentArgs&)+232) (BuildId: 0a10c9cc500ee91d45b0d2d080c482eb)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #03 pc 0000000000044ae0  /system_ext/lib64/libcomposerextn.qti.so (smomo::SmomoImpl::FrameIsLate(int, long)+808) (BuildId: 0a10c9cc500ee91d45b0d2d080c482eb)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #04 pc 00000000001be204  /system/bin/surfaceflinger (_ZNSt3__110__function6__funcIZN7android14SurfaceFlinger19setTransactionStateERKNS2_17FrameTimelineInfoERKNS2_6VectorINS2_13ComposerStateEEERKNS7_INS2_12DisplayStateEEEjRKNS2_2spINS2_7IBinderEEERKNS2_19InputWindowCommandsElbRKNS2_14client_cache_tEbRKNS_6vectorINS2_17ListenerCallbacksENS_9allocatorISS_EEEEmE4$_64NST_ISY_EEFvRKNS2_13layer_state_tEEEclES12_$316ab8feed68e3a1dff713cb63dee1c6+240) (BuildId: d26adbd9527290a0292af3d634a1b11c)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #05 pc 000000000019f390  /system/bin/surfaceflinger (android::SurfaceFlinger::setTransactionState(android::FrameTimelineInfo const&, android::Vector<android::ComposerState> const&, android::Vector<android::DisplayState> const&, unsigned int, android::sp<android::IBinder> const&, android::InputWindowCommands const&, long, bool, android::client_cache_t const&, bool, std::__1::vector<android::ListenerCallbacks, std::__1::allocator<android::ListenerCallbacks> > const&, unsigned long)+3176) (BuildId: d26adbd9527290a0292af3d634a1b11c)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #06 pc 00000000000c37c0  /system/lib64/libgui.so (android::BnSurfaceComposer::onTransact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+13468) (BuildId: 19476f020d4c592c5bf1891062fc5de3)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #07 pc 00000000001a85a0  /system/bin/surfaceflinger (android::SurfaceFlinger::onTransact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+820) (BuildId: d26adbd9527290a0292af3d634a1b11c)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #08 pc 000000000004a84c  /system/lib64/libbinder.so (android::BBinder::transact(unsigned int, android::Parcel const&, android::Parcel*, unsigned int)+252) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #09 pc 0000000000054764  /system/lib64/libbinder.so (android::IPCThreadState::executeCommand(int)+1028) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #10 pc 0000000000054294  /system/lib64/libbinder.so (android::IPCThreadState::getAndExecuteCommand()+164) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #11 pc 0000000000054b50  /system/lib64/libbinder.so (android::IPCThreadState::joinThreadPool(bool)+72) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #12 pc 00000000000810d4  /system/lib64/libbinder.so (android::PoolThread::threadLoop()+28) (BuildId: 4d71e7177f5f149ccbd4d7477aa29112)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #13 pc 0000000000013454  /system/lib64/libutils.so (android::Thread::_threadLoop(void*)+264) (BuildId: 5a130e6819cc6e1c6098896371429dc2)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #14 pc 0000000000012cc4  /system/lib64/libutils.so (thread_data_t::trampoline(thread_data_t const*)+408) (BuildId: 5a130e6819cc6e1c6098896371429dc2)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #15 pc 00000000000b508c  /apex/com.android.runtime/lib64/bionic/libc.so (__pthread_start(void*)+264) (BuildId: b94752b6809cd7518c3c28ff9d195306)
02-16 14:31:37.097 13097 13097 F DEBUG   :       #16 pc 0000000000052f9c  /apex/com.android.runtime/lib64/bionic/libc.so (__start_thread+68) (BuildId: b94752b6809cd7518c3c28ff9d195306)
02-16 14:31:37.112 31891 32100 W NativeCrashListener: Couldn't find ProcessRecord for pid 31739
```

@flutter-symbolizer-bot this force flutter#v3.7.3 android release arm64

**Comment by huycozy** [comment]
>[ +328 ms] Service protocol connection closed.
[        ] Lost connection to device.
[        ] DevFS: Deleting filesystem on the device (file:///data/user/0/com.example.webview_flutter_test/code_cache/flutter_webviewCBXKZJ/flutter_webview/)
[ +253 ms] Ignored error while cleaning up DevFS: TimeoutException after 0:00:00.250000: Future not completed
[   +4 ms] "flutter run" took 45,283ms.
[   +3 ms] Running 1 shutdown hook
[  +14 ms] Shutdown hooks complete
[        ] exiting with code 0

From your logs at https://github.com/flutter/flutter/issues/120940#issuecomment-1437990934 when running the [minimal webview sample code](https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter/example/lib/simple_example.dart) and your device crash log at https://github.com/flutter/flutter/issues/120940#issue-1588614200, it is quite similar to https://github.com/flutter/flutter/issues/104889. But #104889 is about `the webview crashes when embedded in a scrollview` whilst this issue is happen randomly on a specific device that I also can not reproduce it from my end.

So, I will label this for further investigation. Once you found some clearer steps to reproduce, please update it in the comments to help us can reproduce it. Thanks!

**Comment by huycozy** [comment]
@stuartmorgan He tried running the [Android Native WebView sample code](https://github.com/huycozy/Android-Native-Webview) (at https://github.com/flutter/flutter/issues/120940#issuecomment-1439577590) but the crash didn't occur.
